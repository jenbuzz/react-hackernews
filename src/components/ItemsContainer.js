import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { getAll } from '../services/api';
import Item from './Item';

const ItemsContainer = ({ type, limit }) => {
    const [items, setItems] = useState([]);
    const [isLoading, setIsLoading] = useState(true);

    useEffect(() => {
        getAll(type, limit).then(items => {
            setItems(items);
            setIsLoading(false);
        });
    }, []);

    return (
        <div className="react-hackernews">
            {isLoading && <span>Loading ...</span>}
            {!isLoading && items.map(item => <Item key={item.id} item={item} />)}
        </div>
    );
};

ItemsContainer.propTypes = {
    type: PropTypes.oneOf(['new', 'top', 'best', 'ask', 'show', 'job']),
    limit: PropTypes.number,
};

ItemsContainer.defaultProps = {
    type: 'top',
    limit: 10,
};

export default ItemsContainer;
