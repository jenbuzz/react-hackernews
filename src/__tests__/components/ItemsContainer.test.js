import React from 'react';
import { shallow } from 'enzyme';
import ItemsContainer from '../../components/ItemsContainer';

describe('ItemsContainer', () => {
    it('renders', () => {
        const component = shallow(<ItemsContainer />);

        expect(component).toMatchSnapshot();
    });
});
