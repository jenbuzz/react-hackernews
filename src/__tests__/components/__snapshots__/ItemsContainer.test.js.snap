// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemsContainer renders 1`] = `
<div
  className="react-hackernews"
>
  <span>
    Loading ...
  </span>
</div>
`;
